### DISCUSSION TITLE
Access the translations within a helper function

### DISCUSSION THREAD

**Discussion by lorezyra** [hexojs/hexo#5190] [Question / Answer] [answered]
How should one use [`i18n`](https://hexo.io/docs/internationalization) inside a helper function?

Ideally, I'd like to use the `__()` function call to get the translated text from the language files.
I'm building a theme and using helpers to build the HTML instead of embedding all the logic in the EJS template. What am I missing?

![Debugging helper](https://user-images.githubusercontent.com/96296311/232907511-5d527b4a-7ee5-4f36-b42f-5ec54486af3b.png)

**Comment by stevenjoezhang** [comment, marked as answer]
Use `this.__`. For example:

```js
hexo.extend.helper.register('post_edit', function(src) {
  const { post_edit } = this.theme;
  if (!post_edit.enable) return '';
  return this.next_url(post_edit.url + src, '<i class="fa fa-pen-nib"></i>', {
    class: 'post-edit-link',
    title: this.__('post.edit')
  });
});
```

**Reply by lorezyra** [reply]
I've tried both `this.__('post.edit')` and `__('post.edit')`. I still get the error: "__ is not a function". Maybe I'm too sleepy to figure this out right now. Time for 🛌💤. Perhaps, I'm not in the right scope.

**Reply by lorezyra** [reply]
Yeah, that was my problem - Scope. Thanks. It works.
